src/mds/LogSegment.h

```diff
@@ -52,8 +74,8 @@ class LogSegment {
   elist<CInode*>  dirty_dirfrag_nest;
   elist<CInode*>  dirty_dirfrag_dirfragtree;
 
-  elist<MDSlaveUpdate*> slave_updates;
-  
+  elist<MDSlaveUpdate*> slave_updates = 0; // passed to begin() manually
+
```

Comment:
I think better call the constructor explicitly with `0` size.
`elist<MDSlaveUpdate*> slave_updates(0);`

Reply:
why?

Reply:
It aligns with https://google.github.io/styleguide/cppguide.html#Implicit_Conversions

Reply:
I have updated the patch.

Reply:
I didn't mean to revert. See https://github.com/ceph/ceph/pull/30202#discussion_r321714580